src/WTG.Analyzers/Analyzers/LinqEnumerable/LinqEnumerableUtils.cs

```diff
@@ -0,0 +1,193 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using WTG.Analyzers.Utils;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace WTG.Analyzers.Analyzers.LinqEnumerable
+{
+	public static class LinqEnumerableUtils
+	{
+		public static InitializerExpressionSyntax? GetInitializer(ExpressionSyntax? e)
+		{
+			return e?.Kind() switch
+			{
+				SyntaxKind.ObjectCreationExpression => ((ObjectCreationExpressionSyntax)e).Initializer,
+				SyntaxKind.ArrayCreationExpression => ((ArrayCreationExpressionSyntax)e).Initializer,
+				SyntaxKind.ImplicitArrayCreationExpression => ((ImplicitArrayCreationExpressionSyntax)e).Initializer,
+				_ => throw new NotImplementedException(), // this should never happen - the analyzer should only look for lists, strongly typed arrays, and implicitly typed arrays
+			};
+		}
+
+		public static ExpressionSyntax? GetValue(ExpressionSyntax? e)
+		{
+			var initializer = GetInitializer(e);
+
+			return initializer?.Expressions.First();
+		}
+
+		public static SyntaxNode FixConcatWithAppendMethod(MemberAccessExpressionSyntax m)
```

Comment:
These aren't really utils, can we move the fixer methods to the CodeFixProvider?